### ISSUE TITLE
(prism-expand-list) is very broken

### ISSUE THREAD

**Issue by markasoftware** [alphapapa/prism.el#4]
For example:

`(prism-expand-list 4 '(1 2 3)) ;; => (1 1 2 2 3 3 3)`

The problem is in how `repeat-n` is determined. Here's my proposed solution, but I'm not sure if it's exactly what you want, because it repeats the last element potentially many times, while the documentation says it will be repeated only one time.

```lisp
(defun prism-expand-list (new-length list)
  "Return LIST expanded to NEW-LENGTH.
Each element of LIST is repeated an equal number of times, except
that the last element may be repeated an extra time when
necessary."
  (let* ((length (length list))
         (_longer-p (or (> new-length length)
                        (user-error "NEW-LENGTH must be longer than LIST")))
         (repeat-n (/ new-length length))
         (final-element-repeat-n (mod new-length length))
	 (final-elements (and (< 0 final-element-repeat-n)
			      (-repeat final-element-repeat-n (-last-item list))))
         (new-list (->> list
                        (--map (-repeat repeat-n it))
                        (-flatten))))
    (if final-elements
        (apply #'-snoc new-list final-elements)
      new-list)))
```

**Comment by alphapapa** [comment]
Yes, it should be rewritten.  It works properly for the particular cases I've used in my config, so I stopped there.  This whole package is still somewhat experimental.  :)

> Here's my proposed solution, but I'm not sure if it's exactly what you want, because it repeats the last element potentially many times, while the documentation says it will be repeated only one time.

The docstring says what the code *should* do; the last element should be repeated, at most, one more time than the other elements are repeated.  That's necessary to give the desired effect when configuring the faces.

I'll look at this sometime, but feel free to suggest other approaches in the meantime.  Thanks for reporting.

**Comment by markasoftware** [comment]
> The last element should be repeated, at most, one more time than the other elements are repeated.

This isn't mathematically possible for arbitrary lengths. If the list is 3 long but requested length is 5, for example, how do you want it to work?

**Comment by alphapapa** [comment]
In that case, it wouldn't be repeated.  The expanded sequence would be, e.g. `1 1 2 2 3`.
